<img align="right" src="https://ds-cs-images.s3.ap-northeast-2.amazonaws.com/Codestates_Fulllogo_Color.png" width=100>

## *DATA SCIENCE / SECTION 2 / SPRINT 3 / NOTE 2*

# 📝 Assignment
---
# Data Wrangling


### 1) 지속적으로 여러분의 포트폴리오 데이터를 랭글링 하세요
- 여러분이 선택한 데이터들을 랭글링하여 모델을 만들기 위한 테이블 형태의 데이터로 만들어 보세요.
- **다음 함수의 result를 완성시키고 리턴값을 과제 제출폼에 제출하세요.**

In [1]:
### 이곳에서 제출폼 과제를 진행해 주세요 ###
import pandas as pd
df = pd.read_csv("https://ds-lecture-data.s3.ap-northeast-2.amazonaws.com/assignments/purchase_data.csv", sep=",",index_col=0)

In [2]:
## 다음 result 함수 안에 male, female 변수를 계산하세요.(groupby, get_group을 사용하세요.)
## sum_male: Age = 20, Gender = "Male" 그룹의 Price를 합하세요.
## sum_female: Age = 20, Gender = "Female" 그룹의 Price를 합하세요.

def result(df):
    
    gender_age = pd.DataFrame(df.groupby(["Gender", "Age"])["Price"].sum().reset_index())

    sum_male = gender_age[gender_age.Age==20].iloc[0, 2]
    sum_female = gender_age[gender_age.Age==20].iloc[1, 2]
    
    print("sum_male: {}, sum_female: {}".format(sum_male, sum_female))
    return round(sum_male + sum_female)

result(df)

sum_male: 43.730000000000004, sum_female: 260.20000000000005


304

### 답 : 304

### 2) 새로 특징들을 만들어 보세요
- 여러분이 선택한 데이터에서 특징 공학을 수행해 보세요.
- **(다중선택) purchase_data를 랭글링하여 다음 조건을 계산하고 결과가 올바른 항목만 과제 제출폼에 제출하세요.**
    1. 나이 1-20세 그룹(20세 포함)의 평균 Price는 3.101818이다.
    2. 나이 21-50세 그룹의 평균 Price는 3.021559이다.
    3. 가장 많이 구입한 아이템의 ID는 178 이다.
    4. 두번째로 매출을 많이 낸 아이템 ID는 145이다.

In [3]:
df.head(10)

SN  Age                 Gender  Item ID  \
Purchase ID                                                       
0                  Lisim78   20                   Male      108   
1              Lisovynya38   40                   Male      143   
2               Ithergue48   24                   Male       92   
3            Chamassasya86   24                   Male      100   
4                Iskosia90   23                   Male      131   
5                  Yalae81   22                   Male       81   
6                Itheria73   36                   Male      169   
7              Iskjaskst81   20                   Male      162   
8                Undjask33   22                   Male       21   
9              Chanosian48   35  Other / Non-Disclosed      136   

                                             Item Name  Price  
Purchase ID                                                    
0            Extraction, Quickblade Of Trembling Hands   3.53  
1                                    Frenzied Scimitar   1.56  
2                                         Final Critic   4.88  
3                                          Blindscythe   3.27  
4                                                 Fury   1.44  
5                                            Dreamkiss   3.61  
6               Interrogator, Blood Blade of the Queen   2.18  
7                                        Abyssal Shard   2.67  
8                                            Souleater   1.10  
9                         Ghastly Adamantite Protector   3.58

In [4]:
# 1. 나이 1-20세 그룹(20세 포함)의 평균 Price는 3.101818이다. > True
df[df["Age"] <= 20]["Price"].mean()

3.101818181818182

In [5]:
# 2. 나이 21-50세 그룹의 평균 Price는 3.021559이다.  > True
age_cutoff = (df['Age'] >= 21) & (df['Age'] <50)
df[age_cutoff]["Price"].mean()

3.0215587044534415

In [6]:
# 3. 가장 많이 구입한 아이템의 ID는 178 이다.  > True
df.groupby("Item ID").size().sort_values(ascending=False)

Item ID
178    12
145     9
108     9
82      9
19      8
       ..
104     1
23      1
180     1
91      1
134     1
Length: 183, dtype: int64

In [7]:
# 두 번째로 매출을 많이 낸 아이템 ID는 145이다. > False
df.groupby("Item ID")["Price"].sum().sort_values(ascending=False)

Item ID
178    50.76
82     44.10
145    41.22
92     39.04
103    34.80
       ...  
125     2.00
126     2.00
23      1.99
104     1.93
42      1.75
Name: Price, Length: 183, dtype: float64

## 🔥 도전과제

### 3) 조금 부족하더라도 최소한의 랭글링으로 데이터를 준비하여 간단한 모델을 학습시켜 보세요
- 여러분들이 선택한 데이터를 가지고 간단한 모델을 학습하고 결과를 리포팅 합니다.

In [ ]:
### 이곳에서 과제를 진행해 주세요 ### 